presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -206,6 +206,9 @@ protected void assertNotSupported(String projection, String message)
                 throw failure;
             }
         }
+        catch (AssertionError e) {
+            assertEquals(e.getMessage(), message);
```

Comment:
Should we just ensure PrestoException is thrown?

Reply:
The exception for `RowType`'s duplicate key check is thrown from `TypeSignature` in analyzing stage. It's in common module and could not recognize `PrestoException`. Seems we could not throw a `PrestoException` from common module. What your opinion, is there a better way to handle this situation?